subprojects/docs/src/docs/userguide/more_about_tasks.adoc

```diff
@@ -939,7 +939,7 @@ One last benefit of defining task inputs and outputs is that Gradle can use this
 
 Before a task is executed for the first time, Gradle takes a fingerprint of the inputs. This fingerprint contains the paths of input files and a hash of the contents of each file. Gradle then executes the task. If the task completes successfully, Gradle takes a fingerprint of the outputs. This fingerprint contains the set of output files and a hash of the contents of each file. Gradle persists both fingerprints for the next time the task is executed.
 
-Each time after that, before the task is executed, Gradle takes a new fingerprint of the inputs and outputs. If the new fingerprints are the same as the previous fingerprints, Gradle assumes that the outputs are up to date and skips the task. If they are not the same, Gradle executes the task. Gradle persists both fingerprints for the next time the task is executed.
+Each time after that, before the task is executed, Gradle first checks if the _lastModified_ or _size_ properties of the file have changed. If they have, Gradle takes a new fingerprint of the inputs and outputs. If the new fingerprints are the same as the previous fingerprints, Gradle assumes that the outputs are up to date and skips the task. If they are not the same, Gradle executes the task. Gradle persists both fingerprints for the next time the task is executed.
```

Comment:
Can you convert this part to the "one sentence per line" style? That makes (later) reviews easier.

Reply:
The check for last modified and size happens on a per file basis, not for all inputs and outputs together. Gradle still will determine a fingerprint of a file where lastModified or size did not change - it just will use the one it took the last time.

So I think we should split this up in maybe this paragraph, and a sentence how Gradle determines the fingerprint of a single file.

Reply:
> The check for last modified and size happens on a per file basis, not for all inputs and outputs together. 

Not sure what you mean but this, how can we check it for inputs and outputs together? 

> Gradle still will determine a fingerprint of a file where lastModified or size did not change - it just will use the one it took the last time.

This is a bit confusing. So it won't actually determine the fingerprint, just reuse the last one, right?

> So I think we should split this up in maybe this paragraph, and a sentence how Gradle determines the fingerprint of a single file.

K, let me try that.

Reply:
> Not sure what you mean but this, how can we check it for inputs and outputs together?

We can't, though that is what the current phrasing suggest:
> Gradle first checks if the _lastModified_ or _size_ properties of the file have changed. If they have, Gradle takes a new fingerprint of the inputs and outputs

What is `the file` in this context?

> This is a bit confusing. So it won't actually determine the fingerprint, just reuse the last one, right?

Yes, it will re-use the last one.

Reply:
Thanks, updated.